File: 删除排序数组中的重复项.cpp
#include<stdio.h>
int removeDuplicates(int* nums, int numsSize) {
    int i,j,k;
    for(i=0,j=1;j<numsSize;){
        if(nums[j]==nums[i]){
            for(k=i;k<numsSize-1;k++){
                nums[k]=nums[k+1];
            }
            numsSize--;
        }
        else{
            i++;
            j++;
        }
    }
    return numsSize;
}
int main(void){
	int nums[5]={0,1,1,2,2};
	int i;
	int len;
	len=removeDuplicates(nums,5);
	for(i=0;i<len;i++){
		printf("%3d",nums[i]);
	}
}

File: 2020年11月/删除排序数组中的重复项.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        c=0
        for i in range(1,len(nums)):
            if nums[i]!=nums[c]:
                nums[c+1]=nums[i]
                c+=1
        return c+1
